cmd/daemonset-check/run_check.go

```diff
@@ -337,6 +337,14 @@ func generateDaemonSetSpec() *appsv1.DaemonSet {
 		log.Errorln("Error getting ownerReference:", err)
 	}
 
+	ownerPod, err := client.CoreV1().Pods(checkNamespace).Get(context.Background(), ownerRef[0].Name, metav1.GetOptions{})
```

Comment:
Also, because `util.GetOwnerRef` returns the owner of the pod that the code runs on, we can expect `ownerRef` to be the name of the Kuberhealthy pod that spawned this daemonset pod.  Later on when we set the daemonset annotations to `ownerPod.Annotations`, we are actually setting the annotations to whatever the _parent_ Kuberhealthy pod has. 

I think what we want here is to set the annotations and owner reference to whatever the daemonset check pod has.